content/browser/renderer_host/brave_navigation_controller_impl.cc

```diff
@@ -0,0 +1,44 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/content/browser/renderer_host/brave_navigation_controller_impl.h"
+
+#include "brave/components/constants/webui_url_constants.h"
+#include "build/build_config.h"
+#include "content/public/common/url_constants.h"
+
+namespace content {
+
+BraveNavigationControllerImpl::BraveNavigationControllerImpl(
+    BrowserContext* browser_context,
+    FrameTree& frame_tree,
+    NavigationControllerDelegate* delegate)
+    : NavigationControllerImpl(browser_context, frame_tree, delegate) {}
+
+BraveNavigationControllerImpl::~BraveNavigationControllerImpl() {}
+
+base::WeakPtr<NavigationHandle>
+BraveNavigationControllerImpl::LoadURLWithParams(const LoadURLParams& params) {
+  if (params.url.SchemeIs(content::kChromeUIUntrustedScheme) &&
+      params.url.host() == kChatUIHost) {
+#if BUILDFLAG(IS_ANDROID)
+    if (!ui::PageTransitionTypeIncludingQualifiersIs(
```

Comment:
I feel it would be helpful to have an explanation why android code is handled differently here.

Reply:
thanks, fixed at https://github.com/brave/brave-core/pull/20996/commits/f840e5dc3a8a356859d6cd721ee8345ad7f19a85#diff-fc1feda162362111f94891cf8248d4682440d13c393d7cdb1e600178ea16b230R24